index.html

```diff
@@ -2697,15 +2697,46 @@ <h3><dfn>Writing or pushing content</dfn></h3>
           |message:NDEFMessageSource| run these steps:
         </p>
         <ol class=algorithm id="create-web-nfc-message">
-          <li>
-            If the |message:NDEFMessageSource| parameter is not of type defined
-            by the <a>NDEFMessageSource</a> union, throw a {{TypeError}} and
-            abort these steps.
-          </li>
-          <li>
-            If the |message:NDEFMessageSource| parameter is of
-            <a>NDEFMessageInit</a> type, and |message|'s records
-            [= list/is empty =], throw a {{TypeError}} and abort these steps.
+          <li>Switch on |message:NDEFMessageSource|'s type:
+            <dl>
+              <dt>{{DOMString}}</dt>
+              <ul>
+                <li>
+                  Let |textRecord| be an <a>NDEFrecord</a> initialized with its
+                  |recordType| set to "`text`" and |data| set to |message|.
+                </li>
+                <li>
+                  Set |message|'s records to a list of records that contain only
```

Comment:
Why can't there be multiple records? This way of writing this also sounds weird, took me a while to understand what you were trying to do

Reply:
This prose is only for `writer.push("foo")` case. So only one text record if I'm not mistaken.

Reply:
@kenchris I guess we're good then?